ce/ce/amf/demands.ts

```diff
@@ -298,33 +290,27 @@ export class Unless extends DemandBlock implements AlternativeFulfillment {
           }
         })) {
           // we found something that looks promising.
-          let filtered = <any>{ $0: item };
-          this.setData(filtered);
-          if (this.run) {
+          this.discoveredData = { $0: item.toString() };
+          const run = this.run?.replace('$0', item.toString());
 
-            const commandline = cmdlineToArray(this.run.replace('$0', item.toString()));
-            const result = await execute(resolve(commandline[0]), commandline.slice(1));
-            if (result.code !== 0) {
+          if (run) {
+            const result = await this.runCmd(run);
```

Comment:
why are we running binaries? what part of the spec does this implement?

Reply:
Demands can run commands, for example to get a version string of a tool.

Reply:
What?! Is there a spec for such a thing?